### ISSUE TITLE
Variable used but not assigned should not be reported for Len() or LenB()

### ISSUE THREAD

**Issue by comintern** [rubberduck-vba/Rubberduck#2010]
Noticed this while poking around on #2004.  These two functions do not require the variable to be initialized and never use the value of the variable passed.  Typical use cases are to size a buffer, pass a buffer size to an API function, or in the case of `Len`, to find the length of a string.  

Of those 3, only the last usage would really make it relevant if the variable was assigned or not. But even in that case it wouldn't signify an _error_, as it would simply function as intended if passed an unassigned String by returning a length of 0.

About the only way to create unintentional errors with `Len` and `LenB` that I'm aware of (other than using `Len` with a `String` in a Unicode context) is to pass them a `Variant`.  These get implicitly cast to a `String` regardless of the underlying type:

```VB
Public Sub Example()
    Dim foo As Variant
    Dim bar As Integer
    
    foo = 123
    bar = 123
    Debug.Print TypeName(foo)   '<-- returns Integer
    Debug.Print TypeName(bar)   '<-- returns Integer
    Debug.Print Len(foo)        '<-- returns 3
    Debug.Print Len(bar)        '<-- returns 2
    Debug.Print LenB(foo)       '<-- returns 6
    Debug.Print LenB(bar)       '<-- returns 2
End Sub
```

If anything _that_ should be the only inspection (something like "Variant is implicitly cast to String") that is reported for these functions.

**Comment by bajabaq** [comment]
I'm not sure if it is the same issue, but when using variable inside of "Right()" function, the never used, never assigned error is encountered (maybe because I call this function by clicking on a shape):

`Public Sub Example()`
`Dim btnName As String`
`btName = ActiveSheet.Shapes(Application.Caller).Name 'eg Operate1`           
`Dim d As String`
`d = Right(btnName, 1) '1, 2, or 3`
`MsgBox d`
`End Sub`

**Comment by retailcoder** [comment]
@bajabaq use `Option Explicit`, you have a typo - `btName` isn't declared anywhere. Rubberduck correctly resolves the reference to `btnName` inside the `Right(btnName, 1)` statement, and is correctly warning you about `btnName` not being assigned *and* being used without being assigned.

To be clear, you have:

    btName = ActiveSheet.Shapes(Application.Caller).Name 'eg Operate1

And you need:

    btnName = ActiveSheet.Shapes(Application.Caller).Name 'eg Operate1

**Comment by bajabaq** [comment]
Fixed typo in my code - I am using Option Explicit, and all variables match. I tried to test further by installing 2.08, but it hangs (see issue [2252](https://github.com/rubberduck-vba/Rubberduck/issues/2252)) - and now I can't get back to 1.43 to dig deeper.

Thank you for developing RubberDuck - it is something sorely needed.

**Comment by retailcoder** [comment]
> Thank you for developing RubberDuck - it is something sorely needed.

@bajabaq Thanks for your support! v1.4.3 has *several* parsing and resolving problems that v2.x fixes. Just to be clear, which version is reporting `btnName` as "not used/assigned"? I can't reproduce the issue with the current build.

Rubberduck sorely needs to clean up its COM interactions so 2.x can stop randomly crashing - that's the only reason we haven't released a "stable" 2.0 yet... I haven't had much time to devote to the project lately, but I hope to be able to get back to it soon.

**Comment by bajabaq** [comment]
It was version 1.43 - which was the reason I switched to 2.0.8.
